engine/sync_manager.go

```diff
@@ -249,33 +251,42 @@ func (m *syncManager) Stop() error {
 	return nil
 }
 
-func (m *syncManager) get(exchangeName string, p currency.Pair, a asset.Item) (*currencyPairSyncAgent, error) {
+func (m *syncManager) getWithLock(exchangeName string, p currency.Pair, a asset.Item) (*currencyPairSyncAgent, error) {
 	m.mux.Lock()
 	defer m.mux.Unlock()
+	return m.getNoLock(exchangeName, p, a)
+}
 
-	for x := range m.currencyPairs {
-		if m.currencyPairs[x].Exchange == exchangeName &&
-			m.currencyPairs[x].Pair.Equal(p) &&
-			m.currencyPairs[x].AssetType == a {
-			return &m.currencyPairs[x], nil
-		}
+func (m *syncManager) getNoLock(exchangeName string, p currency.Pair, a asset.Item) (*currencyPairSyncAgent, error) {
```

Comment:
If you're going to be introducing these style `noLocks` functions around the place, they need commentary